### PR TITLE
Java: Fix QLDoc for `Container.toString()`

### LINKED ISSUE #5828: Java: `CompilationUnit.toString()` violates `Container.toString()` API contract
The [documentation](https://codeql.github.com/codeql-standard-libraries/java/semmle/code/FileSystem.qll/predicate.FileSystem$Container$toString.0.html) for `Container.toString()` says:
> Gets a textual representation of the path of this container.
>
> This is the absolute path of the container.

`CompilationUnit` extends `File` (which extends `Container`), however its `toString()` predicate violates the API contract shown above. Instead of having the absolute path as result (e.g. `mypackage/MyClass.java`), it only has the name of the compilation unit as result (e.g. `MyClass`).

Currently `CompilationUnit.toString()` does not overwrite the inherited QLDoc either so its [documentation](https://codeql.github.com/codeql-standard-libraries/java/semmle/code/java/CompilationUnit.qll/predicate.CompilationUnit$CompilationUnit$toString.0.html) also erroneously claims that it has the absolute path as result.

The cleanest solution might be to simply change the documentation for `Container.toString()` since queries should avoid relying on `toString()` anyways (see also #5817). However, since for `CompilationUnit` the result is not even a path, it might be best if the QLDoc simply inherited the default `Top.toString()` text.

### PR DESCRIPTION
**Pull request by dbartol**
Fixes #5828

The QLDoc was just too specific about the default implementation. I've improved the wording.

### CODE REVIEW

File: java/ql/src/semmle/code/FileSystem.qll [hunk 1]
```diff
@@ -146,9 +146,11 @@ class Container extends @container, Top {
   }
 
   /**
-   * Gets a textual representation of the path of this container.
+   * Gets a textual representation of this container.
    *
-   * This is the absolute path of the container.
+   * The default implementation returns the absolute path to the container, but subclasses may
+   * may override to provide a different result. To get the absolute path for any `Container`, call
```
Comment (Marcono1234) [inline java/ql/src/semmle/code/FileSystem.qll:152]: The sentence contains "may" twice:
```suggestion
   * override to provide a different result. To get the absolute path for any `Container`, call
```

Maybe there is also something missing after "override"? (might need line wrapping)
```suggestion
   * override this predicate to provide a different result. To get the absolute path for any `Container`, call
```

File: java/ql/src/semmle/code/FileSystem.qll [hunk 2]
```diff
@@ -146,9 +146,11 @@ class Container extends @container, Top {
   }
 
   /**
-   * Gets a textual representation of the path of this container.
+   * Gets a textual representation of this container.
    *
-   * This is the absolute path of the container.
+   * The default implementation returns the absolute path to the container, but subclasses may
```
Comment (Marcono1234) [inline java/ql/src/semmle/code/FileSystem.qll:151]:
```suggestion
   * The default implementation returns the absolute path of the container, but subclasses may
```
The previous text also said "path **of** this container", though maybe "to the" is also fine.

**Review by Marcono1234 — COMMENTED** [review]
Thanks! Sounds good to me.

File: java/ql/src/semmle/code/FileSystem.qll (hunk 2 above)
Comment (Marcono1234) [reply java/ql/src/semmle/code/FileSystem.qll:151]: As but maybe "returns" needs to be rephrased (e.g. "has ... as result") to conform to the [QLDoc style guide](https://github.com/github/codeql/blob/main/docs/qldoc-style-guide.md#predicates-with-result)?

**Review by aschackmull — APPROVED** [review]

**Pull request merged**